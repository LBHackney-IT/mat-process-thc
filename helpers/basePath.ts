import basePath from "../server/helpers/basePath";

export default basePath;
