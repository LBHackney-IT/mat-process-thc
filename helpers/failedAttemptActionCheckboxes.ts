export default [
  {
    label: "Calling card left",
    value: "calling card",
  },
  {
    label: "Asked tenant to make contact",
    value: "requested contact",
  },
  {
    label: "Other",
    value: "other",
  },
];
