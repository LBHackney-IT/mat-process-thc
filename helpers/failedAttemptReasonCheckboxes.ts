export default [
  {
    label: "Tenant not in",
    value: "tenant not in",
  },
  {
    label: "Not safe to enter property",
    value: "not safe",
  },
  {
    label: "Property may be abandoned",
    value: "abandoned",
  },
  {
    label: "There are signs of subletting",
    value: "subletting",
  },
  {
    label: "Tenant refuses entry",
    value: "refused",
  },
  {
    label: "Other",
    value: "other",
  },
];
