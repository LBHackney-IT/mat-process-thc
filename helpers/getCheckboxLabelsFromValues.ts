export const getCheckboxLabelsFromValues = (
  checkboxes: { label: string; value: string }[],
  values: string[]
): string => {
  return checkboxes
    .filter(({ value }) => values.includes(value))
    .map(({ label }) => label)
    .join(", ");
};
