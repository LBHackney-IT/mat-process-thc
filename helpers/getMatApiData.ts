import { nullAsUndefined } from "null-as-undefined";
import isServer from "./isServer";

const getMatApiData = (processRef: string | undefined): string | undefined => {
  if (isServer) {
    return;
  }

  if (!processRef) {
    return;
  }

  return nullAsUndefined(sessionStorage.getItem(`${processRef}:matApiData`));
};

export default getMatApiData;
