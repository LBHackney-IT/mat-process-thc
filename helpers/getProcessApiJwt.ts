import { nullAsUndefined } from "null-as-undefined";
import isServer from "./isServer";

const getProcessApiJwt = (
  processRef: string | undefined
): string | undefined => {
  if (isServer) {
    return;
  }

  if (!processRef) {
    return;
  }

  return nullAsUndefined(sessionStorage.getItem(`${processRef}:processApiJwt`));
};

export default getProcessApiJwt;
