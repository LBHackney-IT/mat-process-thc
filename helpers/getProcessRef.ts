import { NextRouter } from "next/router";
import isServer from "./isServer";

const getProcessRef = (router: NextRouter): string | undefined => {
  if (isServer) {
    return;
  }

  // `router.query` might be an empty object when first loading a page for
  // some reason.
  const ref = router.query.processRef;

  let processRef: string | undefined = undefined;

  if (ref) {
    processRef = typeof ref === "string" ? ref : ref.join("/");
  }

  if (router.route && router.route.includes("[processRef]")) {
    processRef = processRef || "placeholder-process-ref";
  }

  if (process.env.NODE_ENV !== "production") {
    processRef = processRef || process.env.TEST_PROCESS_REF;
  }

  return processRef;
};

export default getProcessRef;
