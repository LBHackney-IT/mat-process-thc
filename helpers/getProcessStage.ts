import { NextRouter } from "next/router";
import isServer from "./isServer";

const getProcessStage = (router: NextRouter): string | undefined => {
  if (isServer) {
    return;
  }

  // `router.query` might be an empty object when first loading a page for
  // some reason.
  const ref = router.query.processRef;

  return sessionStorage.getItem(`${ref}:processStage`) || "0";
};

export default getProcessStage;
