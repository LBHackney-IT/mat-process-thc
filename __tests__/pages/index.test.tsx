import isOnline from "is-online";
import React from "react";
import { ReactTestRenderer, act, create } from "react-test-renderer";
import useSWR, { ConfigInterface, keyInterface, responseInterface } from "swr";

import { promiseToWaitForNextTick } from "../helpers/promise";

import IndexPage from "../../pages/index";

jest.mock("is-online");
jest.mock("swr");

const isOnlineMock = (isOnline as unknown) as jest.MockInstance<
  Promise<boolean>,
  [isOnline.Options?]
>;

const useSWRMock = (useSWR as unknown) as jest.MockInstance<
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  responseInterface<any, any>,
  | [keyInterface, ConfigInterface?]
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  | [keyInterface, ((...args: any) => any)?, ConfigInterface?]
>;

it("renders correctly when online", async () => {
  isOnlineMock.mockResolvedValue(true);
  useSWRMock.mockImplementation(() => ({
    data: { online: true },
    revalidate: jest.fn(),
    isValidating: false
  }));

  let component: ReactTestRenderer | undefined = undefined;

  await act(async () => {
    component = create(<IndexPage />);

    await promiseToWaitForNextTick();
  });

  expect(component).toMatchInlineSnapshot(`
    <main
      className="govuk-main-wrapper lbh-main-wrapper"
      id="main-content"
      role="main"
    >
      <div
        className="govuk-container lbh-container"
      >
        <h1
          className="lbh-heading-h1"
        >
          Tenancy and Household Check
        </h1>
        <dl
          className="govuk-summary-list lbh-summary-list govuk-summary-list--no-border"
        >
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Address
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              1 Mare Street, London, E8 3AA
            </dd>
          </div>
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Tenants
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              Jane Doe, John Doe
            </dd>
          </div>
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Tenure type
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              Introductory
            </dd>
          </div>
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Tenancy start date
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              1 January 2019
            </dd>
          </div>
        </dl>
        <h2
          className="lbh-heading-h2"
        >
          Previsit setup
        </h2>
        <p
          className="lbh-body"
        >
          The system is currently updating the information you need for this process so that you can work offline or online.
        </p>
        <p
          className="lbh-body"
        >
          Please wait until the ‘Go’ button is available to be clicked before proceeding.
        </p>
        <button
          aria-disabled={false}
          className="govuk-button lbh-button"
          data-prevent-double-click={true}
          disabled={false}
          onClick={[Function]}
        >
          Go
        </button>
      </div>
    </main>
  `);
});

it("renders correctly when offline", async () => {
  isOnlineMock.mockResolvedValue(true);
  useSWRMock.mockImplementation(() => ({
    error: new Error("Request timed out"),
    revalidate: jest.fn(),
    isValidating: false
  }));

  let component: ReactTestRenderer | undefined = undefined;

  await act(async () => {
    component = create(<IndexPage />);

    await promiseToWaitForNextTick();
  });

  expect(component).toMatchInlineSnapshot(`
    <main
      className="govuk-main-wrapper lbh-main-wrapper"
      id="main-content"
      role="main"
    >
      <div
        className="govuk-container lbh-container"
      >
        <h1
          className="lbh-heading-h1"
        >
          Tenancy and Household Check
        </h1>
        <dl
          className="govuk-summary-list lbh-summary-list govuk-summary-list--no-border"
        >
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Address
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              Loading...
            </dd>
          </div>
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Tenants
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              Loading...
            </dd>
          </div>
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Tenure type
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              Loading...
            </dd>
          </div>
          <div
            className="govuk-summary-list__row lbh-summary-list__row"
          >
            <dt
              className="govuk-summary-list__key lbh-summary-list__key"
            >
              Tenancy start date
            </dt>
            <dd
              className="govuk-summary-list__value lbh-summary-list__value"
            >
              Loading...
            </dd>
          </div>
        </dl>
        <h2
          className="lbh-heading-h2"
        >
          Previsit setup
        </h2>
        <p
          className="lbh-body"
        >
          The system is currently updating the information you need for this process so that you can work offline or online.
        </p>
        <p
          className="lbh-body"
        >
          Please wait until the ‘Go’ button is available to be clicked before proceeding.
        </p>
        <button
          aria-disabled={true}
          className="govuk-button lbh-button"
          data-prevent-double-click={true}
          disabled={true}
          onClick={[Function]}
        >
          Loading...
        </button>
      </div>
    </main>
  `);
});
