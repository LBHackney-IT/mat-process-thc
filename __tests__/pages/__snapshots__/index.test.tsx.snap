// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h1>
  This will be a start page!
</h1>
`;
